sycl/test-e2e/InvokeSimd/Regression/nbarrier_exec_in_order.cpp

```diff
@@ -127,27 +134,53 @@ bool test(QueueTY q) {
     out[i] = -1;
   }
 
+  auto dev = q.get_device();
+  auto deviceSLMSize =
+      dev.template get_info<sycl::info::device::local_mem_size>();
+  if constexpr (UseSLM)
+    std::cout << "Local Memory Size: " << deviceSLMSize << std::endl;
+
+  // The test is going to use Size elements of int type.
+  if (deviceSLMSize < Size * sizeof(int)) {
+    // Report an error - the test needs a fix.
+    std::cerr << "Error: Test needs more SLM memory than device has"
+              << std::endl;
+    return false;
+  }
+
   try {
     // workgroups
     sycl::range<1> GlobalRange{Size};
     // threads in each group
-    sycl::range<1> LocalRange{Threads};
+    sycl::range<1> LocalRange{Size};
     sycl::nd_range<1> Range{GlobalRange * LocalRange, LocalRange};
 
     auto e = q.submit([&](handler &cgh) {
+      auto LocalAcc = local_accessor<int, 1>(Size, cgh);
```

Comment:
The creation and passing to ESIMDS of this local_accessor is done even with UseSLM=false.
Doesn't it look like a test design issue/problem to you?

Reply:
As a workaround, I created a dummy local_accessor with a size of 0 and added guards wherever SLM was used. Is this acceptable?

Reply:
Thank you. It is better, but still odd because UseSLM=false cases would have to be disabled if something happens with passing local accessors (e.g. regression in gpu rt).
I am not insisting on splitting the tests though as it is code duplication. It is up to you.